cudax/test/device/device_smoke.cu

```diff
@@ -35,6 +35,14 @@ TEST_CASE("Smoke", "[device]")
   using cudax::device;
   using cudax::device_ref;
 
+  SECTION("Compare")
+  {
+    CUDAX_REQUIRE(device_ref{0} == device_ref{0});
+    CUDAX_REQUIRE(device_ref{0} == 0);
+    CUDAX_REQUIRE(device_ref{1} != device_ref{0});
+    CUDAX_REQUIRE(device_ref{1} != 2);
```

Comment:
We might also need to check the other way around because that might not work in C++17::
```suggestion
    CUDAX_REQUIRE(device_ref{0} == 0);
    CUDAX_REQUIRE(0 == device_ref{0});
    CUDAX_REQUIRE(device_ref{1} != device_ref{0});
    CUDAX_REQUIRE(device_ref{1} != 2);
    CUDAX_REQUIRE(1 != device_ref{2});
```